Plugins/Spell/Spell.cpp

```diff
@@ -0,0 +1,182 @@
+    
+    if(!handle)
+    {
+        throw std::runtime_error("Dynamic link handler error");
+    }
+    
+    Spell::setcreate = reinterpret_cast<Spell::Create_Exp>(dlsym(handle, "Hunspell_create"));
+    
+    if (!setcreate) 
+    { 
+        throw std::runtime_error("Dynamic link symbol error");
+    }
+    
+    Spell::spell_e = reinterpret_cast<Spell::Spell_Exp>(dlsym(handle, "Hunspell_spell"));
+    if (!spell_e) 
+    { 
+        throw std::runtime_error("Dynamic link symbol error");
+    }
+
+    Spell::suggest_e = reinterpret_cast<Spell::Suggest_Exp>(dlsym(handle, "Hunspell_suggest"));
+    if (!suggest_e) 
+    { 
+        throw std::runtime_error("Dynamic link symbol error");
+    }
+    Spell::dest_e = reinterpret_cast<Spell::Des_Exp>(dlsym(handle, "Hunspell_destroy"));
+    if (!dest_e) 
+    { 
+        throw std::runtime_error("Dynamic link symbol error");
+    }
+    
+    Spell::free_e = reinterpret_cast<Spell::Free_Exp>(dlsym(handle, "Hunspell_free_list"));
+    if (!free_e) 
+    { 
+        throw std::runtime_error("Dynamic link symbol error");
+    }
+    Spell::dic = config->Get<std::string>("PATH_DIC", "/usr/share/hunspell/en_US.dic");
+    Spell::aff = config->Get<std::string>("PATH_AFF", "/usr/share/hunspell/en_US.aff");
+    
+    Spell::created = setcreate(Spell::aff.c_str(), Spell::dic.c_str());
+}
+ArgumentStack Spell::GetSpell(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+    
+    Spell::Init(GetServices()->m_config);
+
+
+    std::string sentence = Services::Events::ExtractArgument<std::string>(args);
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), ','), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), ':'), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), ';'), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), '"'), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), '('), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), ')'), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), ')'), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), '*'), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), '['), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), ']'), sentence.end());
+    sentence.erase(std::remove(sentence.begin(), sentence.end(), '.'), sentence.end());
+    std::istringstream iss(sentence);
+    std::string  word;
+    std::string output = "";
+    int sc;
+    while (getline( iss, word, ' ' ))
+    {
+        if(!word.empty())
+        {
+            sc = Spell::spell_e(Spell::created, word.c_str());
+            if(sc == 0)
+              output += word + ",";         
+        }
+
+        
+    }
+  
+    Spell::dest_e(Spell::created);
+    dlclose(Spell::handle);
```

Comment:
I don't think you want to be closing this after each event - after all, you only open it once, in the Init function. You probably want to close it in the destructor.

Reply:
Oh wait, scratch that, you call Init every time ... is that a good idea? Seems like you should only initialise it once in the constructor, and free it in the destructor.